### ISSUE TITLE
Huber Loss calculation incorrect?

### ISSUE THREAD

**Issue by trickmeyer** [reinforceio/tensorforce#139]
I was noticing unexpected negative losses in my Q learning, and think I've tracked it down.  In q_model.py, the huber_loss is calculated as follows:

`huber_loss = tf.where(condition=(tf.abs(delta) < config.clip_loss), x=(0.5 * self.loss_per_instance), y=(tf.abs(delta) - 0.5))
`

I believe this is incorrect in the `y=` condition (i.e. `tf.abs(delta) >= config.clip_loss` - when the affine/linear portion kicks in).  It seems `config.clip_loss` isn't even taken into account in these cases.  

I believe the correct calculation is as follows (via [Tensorflow](https://www.tensorflow.org/api_docs/python/tf/losses/huber_loss) or [Wikipedia](https://en.wikipedia.org/wiki/Huber_loss)):

```
  0.5 * x^2                  if |x| <= d
  0.5 * d^2 + d * (|x| - d)  if |x| > d
```
Where `x=labels-predictions` and `d` would be `config.clip_loss` in our implementation.  Therefore, I believe the correct implementation should be:

`huber_loss = tf.where(condition=(tf.abs(delta) < config.clip_loss), x=(0.5 * self.loss_per_instance), y=0.5 * config.clip_loss ** 2 + config.clip_loss * (tf.abs(delta) - config.clip_loss))`

Can someone check my work?  I'm happy to create a PR.

**Comment by AlexKuhnle** [comment]
You're right. If you could create a PR, that would be great, thanks!

**Comment by trickmeyer** [comment]
Submitting shortly.  Rearranging terms above for a simpler (to read) calculation:

`config.clip_loss * tf.abs(delta) - 0.5 * config.clip_loss ** 2`

**Comment by AlexKuhnle** [comment]
Closed, thanks for the PR.
